### ISSUE TITLE
team_id supplied to match does not trickle down to its cert and sigh calls if the project has another team_id set

### ISSUE THREAD

**Issue by SpacyRicochet** [fastlane/fastlane#3996]
It looks like the `sigh` and `cert` commands executed by `match` ignore the team identifier supplied to `match` and defaults to the default team identifier in the Appfile instead.

The project is one where one whitelabel app has has a build target per label and each label has its own Apple Developer Team. So I use a `fastlane match` to make sure I have all the certificates necessary for all the labels. 

---

**Appfile**

```
app_identifier "com.companyA.iphonedev"
apple_id "devs@noodlewerk.com"
team_id "AAAAAAAAAA"
```

**Matchfile**

```
git_url "git@our.repository.com:certificates_A.git"
```

**Fastfile**

```
lane :match do
... snipped other labels ...
  match(type: "development", app_identifier: "com.companyB.iphonedev", team_id: "BBBBBBBBBB", git_url: "git@our.repository.com:certificates_B.git")
... snipped other labels ...
end
```

---

**Edited terminal output after performing `fastlane match`**

```
[10:59:39]: -------------------
[10:59:39]: --- Step: match ---
[10:59:39]: -------------------

+-----------------------+------------------------------------------------------+
|                           Summary for match 0.4.0                            |
+-----------------------+------------------------------------------------------+
| type                  | development                                          |
| app_identifier        | com.companyB.iphonedev                               |
| team_id               | BBBBBBBBBB                                           |
| git_url               | git@our.repository.com:certificates_B.git            |
| username              | devs@noodlewerk.com                                  |
| keychain_name         | login.keychain                                       |
| readonly              | false                                                |
| verbose               | false                                                |
| force                 | false                                                |
| shallow_clone         | false                                                |
| force_for_new_devices | false                                                |
| skip_docs             | false                                                |
+-----------------------+------------------------------------------------------+

... snipped ...

+---------------+----------------------------------------------------------+
|                          Summary for cert 1.4.0                          |
+---------------+----------------------------------------------------------+
| development   | true                                                     |
| force         | true                                                     |
| username      | devs@noodlewerk.com                                      |
| team_id       | AAAAAAAAAA                                               |
| keychain_path | /Users/michiel_boertjes/Library/Keychains/login.keychain |
+---------------+----------------------------------------------------------+

... snipped ...

+-------------------------------------+-----------------------------------------+
|                            Summary for sigh 1.6.0                             |
+-------------------------------------+-----------------------------------------+
| app_identifier                      | com.companyB.iphonedev                  |
| adhoc                               | false                                   |
| development                         | true                                    |
| username                            | devs@noodlewerk.com                     |
| force                               | true                                    |
| cert_id                             | 5Z6S9C3FQ7                              |
| provisioning_name                   | match Development com.companyB.iphonedev|
| ignore_profiles_with_different_name | true                                    |
| skip_install                        | false                                   |
| team_id                             | AAAAAAAAAA                              |
| skip_fetch_profiles                 | false                                   |
| skip_certificate_verification       | false                                   |
+-------------------------------------+-----------------------------------------+
```

As you can see, the team identifier for `match` is BBBBBBBBBB as expected, but the for `cert` and `sigh` it's AAAAAAAAAA as mentioned in the `Appfile`.

**Comment by TKBurner** [comment]
Thanks for pointing this out @SpacyRicochet! We are going to look into this :+1:

**Comment by KrauseFx** [comment]
Oh, interesting, that's definitely a bug we need to fix!

**Comment by ch8908** [comment]
Same issue here, plz fix it.

**Comment by TC94615** [comment]
Same issue here.

**Comment by mfurtak** [comment]
I've merged #4355 as a fix. Please give it a try - I'll push it out with the next release 🚀

**Comment by mfurtak** [comment]
Hi folks! #4355 has been shipped as part of [`match` 0.5.0](https://github.com/fastlane/fastlane/releases/tag/match/0.5.0) 🚀
